test/common/router/header_formatter_test.cc

```diff
@@ -527,7 +527,8 @@ TEST_F(StreamInfoHeaderFormatterTest, TestFormatWithUpstreamMetadataVariable) {
                  "false");
 
   // Number values.
-  testFormatting(stream_info, "UPSTREAM_METADATA([\"namespace\", \"nested\", \"num_key1\"])", "1");
+  testFormatting(stream_info, "UPSTREAM_METADATA([\"namespace\", \"nested\", \"num_key1\"])",
```

Comment:
Yeah per @jmarantz I think we would consider this a breaking change. It should at least have release notes, but optimally it wouldn't change at all. Is there a format specifier and/or cast that would make it so that it didn't have to change?

Reply:
There's a `:g` specifier that will remove trailing 0s from doubles. I've added that in the places that tests caught. There's still a risk that doubles will start being printed with trailing 0s where they weren't before if there's no test coverage, though.

Reply:
Yeah that's fair. We will need to go with the test coverage we have. Thank you!